### PR TITLE
WebUI: Fix jQuery DOM manipulation issues

### PR DESCRIPTION
**Pull request by serg-cymbaluk**
The commit includes the following jQuery patches:
- Manipulation: Make jQuery.htmlPrefilter an identity function
  (jquery/jquery#4642)
- Manipulation: Skip the select wrapper for <option> outside of IE 9
  (jquery/jquery#4647)

In addition there is included a script that helps to patch and build
the new version of jQuery:

`  $ install/ui/util/make-jquery.js 3.4.1`

Ticket: https://pagure.io/freeipa/issue/8507

### CODE REVIEW

File: install/ui/util/make-jquery.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    echo '  $ ./make-jquery.sh 3.4.1'
+    echo
+    exit 1
+fi
+
+WD=$(realpath $(dirname "${BASH_SOURCE[0]}"))
+JQUERY_VERSION=$1
+
+# Apply patches
+cd /tmp
```
Comment (abbra) [inline install/ui/util/make-jquery.sh:15]: Please use `mktemp` to create a temporary directory instead. Record its name and remove afterwards.

File: install/ui/util/make-jquery.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+git clone -b ${JQUERY_VERSION} --depth 1 https://github.com/jquery/jquery.git
+cd jquery
+git am ${WD}/jquery-patches/${JQUERY_VERSION}/*
+
+# Build jQuery
+npm install
+npm run-script build
+
+# Replace the project version of jQuery with the built one
+cp -fv dist/jquery.min.js ${WD}/../src/libs/jquery.js
```
Comment (abbra) [inline install/ui/util/make-jquery.sh:25]: You also need to clean up the resulted build tree and sources.

**Review by abbra — CHANGES_REQUESTED** [review]

File: install/ui/util/make-jquery.sh (hunk 1 above)
Comment (serg-cymbaluk) [reply install/ui/util/make-jquery.sh]: Thanks for the remark. Fixed.

File: install/ui/util/make-jquery.sh (hunk 2 above)
Comment (serg-cymbaluk) [reply install/ui/util/make-jquery.sh:25]: Fixed

Comment (abbra) [reply install/ui/util/make-jquery.sh:25]: Please change the name for `$TMPDIR` variable. `$TMPDIR` is actually a reserved name used in the system environment by many utilities to point to the current temporary directory. For example, `mktemp` uses it already.

Comment (serg-cymbaluk) [reply install/ui/util/make-jquery.sh:25]: Changed

**Comment by abbra** [comment]
LGTM

**Comment by serg-cymbaluk** [comment]
master:

* 29b41aef0a49d5460631a8543f0f6620d4696790 WebUI: Fix jQuery DOM manipulation issues

**Pull request closed without merging**